### PR TITLE
[SYCLomatic][Intercept-build] Refine intercept-build tool when tool chain is available

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

**Review by ShengchenJ — APPROVED** [review]

File: clang/tools/scan-build-py/lib/libscanbuild/intercept.py [hunk 1]
```diff
@@ -162,16 +168,83 @@ def post_processing(commands):
         # dump the compilation database
         with open(args.cdb, "w+") as handle:
             json.dump(entries, handle, sort_keys=True, indent=4)
+
+        tmp_file = "/tmp/is_nvcc_available.txt" # temp file is created by libear.so
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libscanbuild/intercept.py]: Avoid use tmp file.

Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libscanbuild/intercept.py]: Okay

File: clang/tools/scan-build-py/lib/libear/ear.c [hunk 2]
```diff
@@ -503,6 +507,63 @@ int eaccess(const char *pathname, int mode) {
   return call_eaccess(pathname, mode);
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libear/ear.c]: L507  no need call again, just return none zero, as the result is there already.

Comment (tomflinda) [reply clang/tools/scan-build-py/lib/libear/ear.c]: Okay, will refine it in the updated PR.

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**